Support avertv cardbus hybrid e506r

Reported in
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=511385

This patch is a subset of the following commit:

From: Massimo Piccioni <alsa@piccio.org>
Date: Fri, 11 Jul 2008 16:48:02 +0000 (-0300)
Subject: V4L/DVB (8244): saa7134: add support for AVerMedia M103
X-Git-Tag: v2.6.27-rc1~966^2~125
X-Git-Url: http://git.kernel.org/?p=linux%2Fkernel%2Fgit%2Ftorvalds%2Flinux-2.6.git;a=commitdiff_plain;h=e2fc00c21124d9d9a8d4f396e5498ea27ddfc8fd

V4L/DVB (8244): saa7134: add support for AVerMedia M103

The following patch updates saa7134 driver to add support for AVerMedia
M103 MiniPCI DVB-T Hybrid card.

Signed-off-by: Massimo Piccioni <alsa@piccio.org>
[mchehab@infradead.org: fixed merge conflicts and a small codingstyle]
Signed-off-by: Mauro Carvalho Chehab <mchehab@infradead.org>
---

diff --git a/drivers/media/video/saa7134/saa7134-dvb.c b/drivers/media/video/saa7134/saa7134-dvb.c
index 341b101..be48b9b 100644
--- a/drivers/media/video/saa7134/saa7134-dvb.c
+++ b/drivers/media/video/saa7134/saa7134-dvb.c
@@ -1263,6 +1263,7 @@ static int dvb_init(struct saa7134_dev *dev)
 						&avermedia_xc3028_mt352_dev,
 						&dev->i2c_adap);
 		attach_xc3028 = 1;
+		break;
 	case SAA7134_BOARD_MD7134_BRIDGE_2:
 		dev->dvb.frontend = dvb_attach(tda10086_attach,
 						&sd1878_4m, &dev->i2c_adap);
